CondFormats/EgammaObjects/interface/GBRTree.h

```diff
@@ -79,61 +79,18 @@
 
 //_______________________________________________________________________
 inline double GBRTree::GetResponse(const float* vector) const {
-  
-  int index = 0;
-  
-  unsigned char cutindex = fCutIndices[0];
-  float cutval = fCutVals[0];
-  
-  while (true) {
-     
-    if (vector[cutindex] > cutval) {
-      index = fRightIndices[index];
-    }
-    else {
-      index = fLeftIndices[index];
-    }
-    
-    if (index>0) {
-      cutindex = fCutIndices[index];
-      cutval = fCutVals[index];
-    }
-    else {
-      return fResponses[-index];
-    }
-    
-  }
-  
-
+  return fResponses[TerminalIndex(vector)];
 }
 
 //_______________________________________________________________________
 inline int GBRTree::TerminalIndex(const float* vector) const {
-  
-  int index = 0;
-  
-  unsigned char cutindex = fCutIndices[0];
-  float cutval = fCutVals[0];
-  
-  while (true) {
-    if (vector[cutindex] > cutval) {
-      index = fRightIndices[index];
-    }
-    else {
-      index = fLeftIndices[index];
-    }
-    
-    if (index>0) {
-      cutindex = fCutIndices[index];
-      cutval = fCutVals[index];
-    }
-    else {
-      return (-index);
-    }
-    
-  }
-  
-
+   int index = 0;
+  do {
+     auto r = fRightIndices[index];
+     auto l = fLeftIndices[index];  
+    index =  vector[fCutIndices[index]] > fCutVals[index] ? r : l;
```

Comment:
yes,
because no where is explicitly said that if
vector[fCutIndices[index]] > fCutVals[index] is true fLeftIndices[index] exists

obvious in
a = p ? p->k : 0
here you cannot evaluate the two options as if p il null the first will segfault

indeed a tree could be implemented in some strange way that not all right and left node exists for a given index...